### ISSUE TITLE
ConfigurationProviders may miss mutliple notifications

### ISSUE THREAD

**Issue by ajcvickers** [aspnet/Extensions#617]
For example, see CreatingOptionalFileInNonExistentDirectoryWillReload. In this test there is a missing directory which is then created and a file is created in it. When this is run without any delays it can miss the notification of the file creation entirely. That is, no matter how long we wait, it will never happen. I haven't fully root-caused this, but it appears to be partly an issue with how change tokens are processed and then re-created asynchronously. It appears this can sometimes leave the application with the wrong change token, or can leave configuration not waiting on the current change token.

**Comment by ajcvickers** [comment]
@HaoK and I discussed this in detail and looked at the code. We're going to close for now as won't fix based on:
* It seems to always work on Windows
* It is most flaky on Mac, where the underlying file system watcher is known to be very unreliable
* As far as we can tell, the issues seen here are entirely due to the unreliable file system watcher
